### PR TITLE
NIFI-1580 - Allow double-click to display config

### PR DESCRIPTION
**Pull request by yuri1969**
* Added double-click shortcut opening config/details dialog to processors,
connections, ports and labels.
* Created a base for further default action selection, disabling, etc.
* Omitted default action configuration UI - that might be a separate JIRA ticket.

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-graph.js [hunk 1]
```diff
@@ -198,13 +201,13 @@
     var nfGraph = {
         init: function () {
             // initialize the object responsible for each type of component
-            nfLabel.init(nfConnectable, nfDraggable, nfSelectable, nfContextMenu);
+            nfLabel.init(nfConnectable, nfDraggable, nfSelectable, nfContextMenu, nfQuickSelect);
             nfFunnel.init(nfConnectable, nfDraggable, nfSelectable, nfContextMenu);
-            nfPort.init(nfConnectable, nfDraggable, nfSelectable, nfContextMenu);
+            nfPort.init(nfConnectable, nfDraggable, nfSelectable, nfContextMenu, nfQuickSelect);
             nfRemoteProcessGroup.init(nfConnectable, nfDraggable, nfSelectable, nfContextMenu);
```
Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-graph.js]: What about the RPG? Shouldn't a user be able to double click to open the config?

Comment (yuri1969) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-graph.js]: I haven't thought about RPGs (the ticket mentions _processors_ only). So RPGs should feature this functionality too, right?

Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-graph.js]: Yea I think so....I mean...what other options are there for a double click on an RPG?

Comment (yuri1969) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-graph.js]: @scottyaslan OK, I'll enable configure by double clicking for RPG.

**Comment by scottyaslan** [comment]
Reviewing...

**Comment by scottyaslan** [comment]
@yuri1969 everything looks good but I noticed that when double clicking on a connection nothing happens. What I would expect to happen is that the 'Connection Configuration Dialog' should open. I think a quickSelect needs to be activated somewhere in the nf-connection.js when a connection is double clicked...

**Comment by yuri1969** [comment]
@scottyaslan It seems I omitted `quickSelect` activation for both start and end "nodes". Please, can you confirm that a double-click displays the connection configuration dialog when performed on a connection's mid "node"?

**Comment by scottyaslan** [comment]
I can confirm that currently you can dblclick on a connection's stats box to open the config/view config. You can also (currently) double click any bends in a connection to open the config/view config for that connection.

**Comment by yuri1969** [comment]
@scottyaslan Thanks. So the problem is just with the start and end nodes, right?

**Comment by scottyaslan** [comment]
hmmmm I see the problem now...I was thinking the issue is with the actual "line". See, If you right click on the connection's 'line' you get the context menu and can then choose to config/view config so I was thinking we should add that 'quickSelect' open config/view config to the 'line' as well as the 'start', 'end', or 'middle' nodes. The problem with this is that we already have an action for double click on the 'line' (that action is for adding a bend). So maybe @moranr could chime in here....

Currently the way it is with this PR you can double click the the stats box or an existing bend on a connection's 'line' to open the config dialog for the connection. However, if you double click on the 'line' you will add a bend to the connection rather than opening the connections' config dialog. Is this the expected behavior? Or, do we want to change the current behavior such that when you double click on a 'line' for a connection you will open the connection's config dialog (just like double click everywhere else on a connection) and then we can add an action to the context menu to 'Add Bend' to make it more explicit to the user....thoughts?

**Comment by yuri1969** [comment]
- I find it natural to double-click on the 'line' when I want to add a new bend.
- In order to check connection's config I would try to double-click on its rectangular stats box.
- Connection's 'nodes' being able to open the config dialog feels OK to me. However, I wouldn't double-click on them intuitively.

I have got a Mule Anypoint Studio background so my opinion might be skewed.

**Comment by moranr** [comment]
I would suggest leaving all current functionality for adding/removing bends, and only add a new double-click action to open the _Configure Connection_ dialog from the stat box.

**Comment by pvillard31** [comment]
+1 - I'd not change the current behaviour for bends.

**Comment by scottyaslan** [comment]
Thanks @yuri1969 this has been merged to master!

**Comment by yuri1969** [comment]
@scottyaslan OK, but I'm afraid the merged version still has the `quickSelect` activated for mid nodes/bends. It seems this should be removed.

**Comment by scottyaslan** [comment]
Sorry about that @yuri1969 I thought we reached consensus that this was expected as long as the current behavior for adding a bend with the double click stayed in tact...

**Comment by pvillard31** [comment]
@yuri1969 - is there a way to submit a PR to revert the behavior when deleting a bend point? Double-click on a bending point opens the connection's configuration instead of deleting it.

**Comment by yuri1969** [comment]
@pvillard31 - I will submit a new PR to revert the current bend points behavior later today or tomorrow.

**Comment by yuri1969** [comment]
@pvillard31 - The reverting PR is #2064.

**Pull request closed without merging**